google/auth/transport/requests.py

```diff
@@ -313,6 +314,9 @@ def my_cert_callback():
             refreshing credentials. If not passed,
             an instance of :class:`~google.auth.transport.requests.Request`
             is created.
+        default_host (Optional[str]): A host like "pubsub.googleapis.com".
```

Comment:
I have a little bit concern about setting the `default_host` in ctor. There are 2 default endpoints, for instance, `pubsub.googleapis.com` and `pubsub.mtls.googleapis.com`. mtls has auto switch logic to switch from the regular endpoint to mtls endpoint. It works as follows:

(1) client creates an AuthorizedSession
(2) client calls AuthorizedSession.is_mtls (with other conditions like the GOOGLE_API_USE_CLIENT_CERTIFICATE env var) to decide which endpoint to use

So this means in the AuthorizedSession ctor, we may not know exactly which default endpoint will be used. So if both endpoints use the regular endpoint as the audience, then this code works fine. Otherwise, we need to find a way. 

@bshaffer

Reply:
mtls only works with user credentials, so this should be fine since service account credentials will be rejected by the server anyway.